.github/workflows/ci.yml

```diff
@@ -0,0 +1,68 @@
+name: ci
+
+# this build will only trigger automatically when `master` pushes
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  ci:
+    name: test-${{ matrix.os }}-python${{ matrix.python-version }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+
+        # this works like a loop; think `pytest.mark.parametrize`
+        python-version: [2.7, 3.5, 3.6, 3.7, 3.8]
+        os: [ubuntu-latest, windows-latest, macos-latest]
+
+    steps:
+    - name: checkout build
+      id: checkout-build
+      uses: actions/checkout@v2
+
+    # with this we do not need to worry about the various ways in which
+    # each os installs python
+    - name: setup python
+      id: setup-python
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: upgrade pip
+      id: upgrade-pip
+      run: |
+        python -m pip install --upgrade pip
+
+    # gets the path to wherever each os stores pip's cache
+    - name: get pip cache dir
+      id: get-pip-cache-dir
+      run: |
+        echo "::set-output name=dir::$(pip cache dir)"
+
+    - name: cache pip
+      uses: actions/cache@v2
+      id: cache-pip
+      with:
+
+        # set above with `get pip cache dir`
+        path: ${{ steps.get-pip-cache-dir.outputs.dir }}
+
+        # determines whether the hash of setup.py has changed
+        # CACHE_VERSION: slight hack, see https://stackoverflow.com/a/64819132
+        key: ${{ runner.os }}-pip-${{ secrets.CACHE_VERSION }}-${{ hashFiles('**/setup.py') }}
+
+    - name: install packages
+      id: install-packages
+      run: |
+        python -m pip install .[dev] codecov
```

Comment:
Would it make sense to use `pipenv` here? Since the deps in `setup.py` aren't pinned

Reply:
I think the problem with pipenv is the strong equality markers. The Pipfile had a range but the lock locks some (all? ) of the packages at a particular point in time. When installing from the lock, some of the packages were not compatible with 2.7. 3.5 (needed to explicitly install another package). So I mirrored the strategy in the travis file. Did you get pipenv working on gitlab?

Reply:
Ahh yeah that's right. I now remember wondering why this project used a `Pipfile.lock`, since (as you noted) the lock breaks across different python versions. 

On GitLab, I was using `tox` to prepare the environment and execute tests, which I believe uses `pip`.